### PR TITLE
feat(open-telemetry-zonneplan): allow ignoring incoming request paths from tracing

### PR DESCRIPTION
**Pull request by Thomas-Vos00**
## Summary

Adds support for excluding certain incoming HTTP request paths (e.g. health/metrics endpoints) from tracing, so they no longer produce server spans.

- New `createDefaultTracingOptions({ ignoredIncomingPaths })` factory that wires an `ignoreIncomingRequestHook` into the HTTP auto-instrumentation.
- `ignoredIncomingPaths` accepts `string` (exact pathname match) or `RegExp` (prefix/pattern match); query strings are stripped before matching.
- `DefaultTracingOptions` is preserved (now defined as `createDefaultTracingOptions()`), so existing consumers are unaffected.
- New public exports: `createDefaultTracingOptions`, `CreateDefaultTracingOptions`, `IgnoredIncomingPath`.

## Example usage

```ts
import { OpenTelemetryBuilder } from '@zonneplan/open-telemetry-node';
import { createDefaultTracingOptions } from '@zonneplan/open-telemetry-zonneplan';

new OpenTelemetryBuilder('my-api')
  .withTracing(
    createDefaultTracingOptions({
      // string => exact pathname match, RegExp => prefix/pattern match
      ignoredIncomingPaths: ['/health', '/metrics', /^\/readiness/]
    })
  )
  .start();
```

The default behaviour is unchanged — `DefaultTracingOptions` (no ignored paths) still works as before:

```ts
import { DefaultTracingOptions } from '@zonneplan/open-telemetry-zonneplan';

new OpenTelemetryBuilder('my-api')
  .withTracing(DefaultTracingOptions)
  .start();
```

## Test plan

- `nx test open-telemetry-zonneplan` — 8 new tests covering exact/regex matching, query-string stripping, empty-paths, and missing-url edge cases.
- `nx lint` / `nx build` pass.

### CODE REVIEW

File: packages/open-telemetry-zonneplan/src/tracing-options.ts [hunk 1]
```diff
@@ -1,20 +1,78 @@
...
+  const { ignoredIncomingPaths = [] } = options;
+
+  return new OpenTelemetryTracingOptionsBuilder()
+    .withInstrumentation(
+      ...getNodeAutoInstrumentations({
+        '@opentelemetry/instrumentation-fs': {
+          enabled: false
+        },
+        '@opentelemetry/instrumentation-http': {
+          ignoreIncomingRequestHook:
```
Comment (Strazz1337) [inline packages/open-telemetry-zonneplan/src/tracing-options.ts:50]: thought: perhaps check whether ignoredIncomingPaths is empty in this method. Otherwise, the ignore hook function is called needlessly on every requested.

Comment (Thomas-Vos00) [reply packages/open-telemetry-zonneplan/src/tracing-options.ts:50]: Good thought — this is already covered: `createIgnoreIncomingRequestHook` returns `undefined` when the paths array is empty (`ignore-incoming-paths.ts:33-35`). So with no `ignoredIncomingPaths` configured we pass `ignoreIncomingRequestHook: undefined`, which means the HTTP instrumentation registers no hook and nothing is called per request. The default `DefaultTracingOptions` therefore has zero per-request overhead. Happy to make that more explicit at the call site if you'd prefer it spelled out here.

Comment (Strazz1337) [reply packages/open-telemetry-zonneplan/src/tracing-options.ts:50]: nvm. This assumption is wrong.

**Review by Strazz1337 — APPROVED** [review]

**Comment by Thomas-Vos00** [comment]
TODO: functioneel testen
